db_test.go

```diff
@@ -145,14 +145,19 @@ func TestSaveExistingDocument(t *testing.T) {
 
 		doc := c.NewDocument()
 		doc.Set(key, "clover")
-		id, err := db.InsertOne(collection, doc)
+		err = db.Save(collection, doc)
 		require.NoError(t, err)
 
+		n, err := db.Query(collection).Count()
+		require.NoError(t, err)
+		require.Equal(t, 1, n)
+
+		id := doc.ObjectId()
 		doc.Set("_id", id)
 		doc.Set(key, updatedValue)
 		require.NoError(t, db.Save(collection, doc))
 
-		n, err := db.Query(collection).Count()
+		n, err = db.Query(collection).Count()
```

Comment:
Same thing here. To be sure that the function is working properly, you need to test if the saved document is equal to the updated one.

```golang
savedDoc, err = db.Query("todos").FindFist()
require.NoError(t, err)
require.Equal(savedDoc)
```